### ISSUE TITLE
pa.sh does not handle arguments correctly

### ISSUE THREAD

**Issue by ko1nksm** [binpash/pash#570]
It seems that pash has a bug in its argument handling.

The script to run is `test.sh`.

```sh
#!/usr/bin/env bash

printf '%s\n' "$@"
```

When executed via `pa.sh`, the space-separated words are the individual arguments.

```console
$  ./test.sh "foo bar" baz
foo bar
baz

$ pa.sh -- ./test.sh "foo bar" baz
foo
bar
baz
```

In addition, pathname expansion is performed.

```sh
$ ls
file1  file2  file3  test.sh

$ pa.sh -- ./test.sh foo '*' baz
foo
file1
file2
file3
test.sh
baz
```

**Comment by angelhof** [comment]
Thanks for reporting these, we will take a look! In general, PaSh has multiple issues with quotes and escaping at the moment, which we are aiming to deal with in a principled way.

**Comment by angelhof** [comment]
These issues are fixed by PR #578 !

**Comment by angelhof** [comment]
Closing since the issues were fixed, but feel free to reopen if you find some other issue related to this, thanks!
